### ISSUE TITLE
Liquidsoap Breaks Command Return on Pipe "|" Character

### ISSUE THREAD

**Issue by SlvrEagle23** [savonet/liquidsoap#1151]
I've been testing to discover the underlying issue behind several reports that certain media doesn't play no matter how it's scheduled, only to discover something very interesting: if the response from the external command includes the pipe "|" character, Liquidsoap appears to interpret that character as starting a new line, so when you pull the first line in the response, it's truncated at that character, i.e.:

Command response:

```
annotate:title="Test Symbol",artist="!@#$%^&*()-=_+[]{}\| Symbol Test":/example/file.mp3
```

...the parsed response from Liquidsoap:

```
annotate:title="Test Symbol",artist="!@#$%^&*()-=_+[]{}\
```

As far as my tests have shown, so far _only_ the pipe character triggers this response. Escaping the pipe character (as is done in the example above) doesn't affect this.

Short of just replacing the character out, any workaround for this?

**Comment by smimram** [comment]
Pipes should _not_ be escaped. I could not reproduce: the following shows `A|B` as artist:
```
s = single("annotate:title=\"bla\",artist=\"A|B\":test.mp3")
s = on_metadata(fun(m) -> log("META: " ^ string_of(m)), s)
out(s)
```

**Comment by SlvrEagle23** [comment]
@smimram I'm not referring to a string being used directly inside the Liquidsoap config, I'm referring to the pipe character being returned from an external process's output:

```
uri = list.hd(get_process_lines("/some/external/process"), default="")
```

^ this will detect the pipe character as an "end of line" indicator, with no apparent way of escaping it to indicate otherwise, so effectively there is no way via `get_process_lines` to actually send the pipe character itself.

**Comment by smimram** [comment]
Ah it must be interpreted by the shell. Did you try \x7C as a way of escaping?

**Comment by SlvrEagle23** [comment]
@smimram If I send back `\x7C` in the response, it simply passes through completely as-is without being converted back into the pipe character, i.e. in the resulting metadata the string `\x7C` is still present.

**Comment by toots** [comment]
I can reproduce:
```
# get_process_lines('echo "aabbcc | ddeff"');;
- : [string] = ["aabbcc ", " ddeff"]
```

**Comment by toots** [comment]
Thanks for the report! It's been fixed and will go out with `1.4.2` very soon.
